src/main/java/de/dennisguse/opentracks/util/EGM2008Utils.java

```diff
@@ -0,0 +1,67 @@
+package de.dennisguse.opentracks.util;
+
+import android.content.Context;
+import android.location.Location;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+
+import de.dennisguse.opentracks.R;
+
+/**
+ * Converts WGS84 altitude to EGM2008 (should be close to height above sea level).
+ * <p>
+ * Uses <a href="https://geographiclib.sourceforge.io/">GeographicLib</a>] EGM2008 5minute undulation data.
+ * https://geographiclib.sourceforge.io/html/geoid.html
+ * <p>
+ * File starts at 90N, 0E (North pole) and is encoded in parallel bands as unsigned shorts.
+ */
+//TODO The altitude correction should be computed using bilinear or cubic interpolation.
+public class EGM2008Utils {
+
+    static final int EGM2008_5_DATA = R.raw.egm2008_5;
+
+    private static final int HEADER_LENGTH = 403;
+
+    private static final int RESOLUTION_IN_MINUTES = 60 / 5;
+
+    private static final int LATITUDE_CORRECTION = 180 * RESOLUTION_IN_MINUTES;
+
+    private EGM2008Utils() {
+    }
+
+    public static double correctedAltitude(Context context, Location location) throws IOException {
```

Comment:
This looks like it is opening the file every time.    I would expect a pattern of opening the file and reading it once - but I am not used to android and it may be better to do more file ops and be less piggy with RAM as you are doing.

Reply:
Actually, it may make sense to keep the file open, saving that time, and still seek to find things.  It would be really good to seek, vs reading and discarding.  (I am of course speaking in POSIX API, not Android..)

Reply:
(I hope) it is not that problematic as we don't need to much undulation data.
Usually, the user will stay in one cell only and we can cache this value in the UI.
If some people track their car/airplane rides that might be different, but we will fix it then.
Then we only do this lookup once. No need to keep the whole file in memory.

And we only get an InputStream from Android to access assets.
Let's see if this works ;)

PS: I need to that only the undulation offset is returned and the getAlltitude() modified by UI only.

Reply:
I guess there are two separate questions.    One is that given a  location do you want to interpolate or just pick closest.   The other is how long to use that location's result -- meaning for other  nearby points -- before refetching.   You could perhaps store [original_location, offset] and then as long as new_location is close use it, else look up again.    It sounds like you are just going to skip that for now  and assume all locations in a  track are close enough - which is still a vast improvement over the old way.